monkey/infection_monkey/master/exploiter.py

```diff
@@ -49,6 +49,27 @@ def exploit_hosts(
             target=self._exploit_hosts_on_queue, args=exploit_args, num_workers=self._num_workers
         )
 
+    @staticmethod
+    def _process_exploiter_config(exploiter_config: Mapping) -> List[Mapping]:
+        # Run vulnerability exploiters before brute force exploiters to minimize the effect of
+        # account lockout due to invalid credentials
+        ordered_exploiters = chain(
+            exploiter_config["vulnerability"], exploiter_config["brute_force"]
+        )
+        exploiters_to_run = list(deepcopy(ordered_exploiters))
```

Comment:
Why do we need to convert an iterable to a list? As far as I can tell, we should be able to use it as it is.

Reply:
Iterators can only be used once, so after the loop on lines 61-64, the iterator would be exhausted and all we could return would be an empty iterator.

Reply:
Oh right. Thanks.